tests/unit/lms/services/grouping_test.py

```diff
@@ -56,7 +84,9 @@ def test_upsert_with_parent_updates_existing_groupings(self, svc, db_session):
         assert db_grouping.extra == new_extra
 
     def test_canvas_group_and_sections_dont_conflict(self, svc, db_session):
-        course = factories.Course(lms_id=self.CONTEXT_ID)
+        course = factories.Course(
+            lms_id=self.CONTEXT_ID, application_instance=svc.application_instance
+        )
```

Comment:
Might this be better as a course fixture which you then set `lms_id` on? It would replace three uses if so, which I think is a good threshold to make a fixture.

Reply:
I think these `Course` objects are only being created by the tests because `svc.upsert_with_parent()` requires the `parent` argument to be a `Course` whose `application_instance` is the same as `svc.application_instance`. If we add a factory or caller method or something (as mentioned in <https://github.com/hypothesis/lms/pull/3444/files#r792004256>) to prevent each test from having to pass every single argument to `upsert_with_parent()` then I think these will go away